### PR TITLE
add create component builder

### PR DESCRIPTION
**Pull request by abreu-ct**
This PR adds the component builder from the adyen connector

### CODE REVIEW

File: enabler/src/components/payment-methods/card/card.ts [hunk 1]
```diff
@@ -1,13 +1,25 @@
-import { ComponentOptions, PaymentMethod } from '../../../payment-enabler/payment-enabler';
+import { ComponentOptions, PaymentComponent, PaymentComponentBuilder, PaymentMethod } from '../../../payment-enabler/payment-enabler';
 import buttonStyles from '../../../style/button.module.scss';
 import inputFieldStyles from '../../../style/inputField.module.scss';
 import styles from '../../../style/style.module.scss';
 import { BaseComponent, BaseOptions } from '../../base';
 import { addFormFieldsEventListeners, fieldIds, getCardBrand, getInput, validateAllFields } from './utils';
 
+export class CardBuilder implements PaymentComponentBuilder {
```
Comment (leungkinghin-ct) [inline enabler/src/components/payment-methods/card/card.ts:8]: In [adyen-integration](https://github.com/commercetools/connect-payment-integration-adyen/blob/main/enabler/src/components/payment-methods/card.ts#L20), it creates `BaseComponentBuilder` and it is inherited by `CardBuilder`. Is there any benefit you found to inherit directly from `PaymentComponentBuilder`?

**Review by leungkinghin-ct — APPROVED** [review]

File: enabler/src/components/payment-methods/card/card.ts (hunk 1 above)
Comment (abreu-ct) [reply enabler/src/components/payment-methods/card/card.ts:8]: Thanks @leungkinghin-ct. I will check it again

Comment (ddeliziact) [inline enabler/src/components/payment-methods/card/card.ts:8]: Lets specify that this component supports the submit... Other option would be to do it on the `base.ts` file to make sure all components support the submit operation.

```suggestion
export class CardBuilder implements PaymentComponentBuilder {
  public componentHasSubmit = true;
```

Comment (abreu-ct) [reply enabler/src/components/payment-methods/card/card.ts:8]: @leungkinghin-ct pls, help me understand the purpose of using a `BaseComponentBuilder` by inheritance which I presume will implement the `PaymentComponentBuilder` interface. Do we want to promote reusing some code?

Which component should we return in the `build`?

```typescript
export abstract class BaseComponentBuilder
  implements PaymentComponentBuilder
{
  public componentHasSubmit = true;

  protected baseOptions: BaseOptions;
  protected paymentMethod: PaymentMethod;

  constructor(paymentMethod: PaymentMethod, baseOptions: BaseOptions) {
    this.paymentMethod = paymentMethod;
    this.baseOptions = baseOptions;
  }

  build(config: ComponentOptions): PaymentComponent {
    // return ?
  }
}
```

Comment (abreu-ct) [reply enabler/src/components/payment-methods/card/card.ts:8]: Thanks @ddeliziact. Fixed it.

Comment (leungkinghin-ct) [reply enabler/src/components/payment-methods/card/card.ts:8]: @abreu-ct I checked the code in adyen repo once again, `BaseComponentBuilder` is used due to supporting multiple payment methods in front-end. For the template repo here, we only use card as the single payment method. Therefore I think it is alright to inherit from `PaymentComponentBuilder`

**Review by leungkinghin-ct — APPROVED** [review]

File: enabler/src/components/payment-methods/card/card.ts (hunk 1 above)
Comment (abreu-ct) [reply enabler/src/components/payment-methods/card/card.ts:8]: Thank you @leungkinghin-ct. If we later on introduce a new payment method we can revisit this.

**Comment by leungkinghin-ct** [comment]
@abreu-ct can you help to resolve the conflicts as well? thank you

**Pull request merged**